### ISSUE TITLE
ADD ability to manually trigger autograde for assignment or for student 

### ISSUE THREAD

**Issue by wabscale** [AnubisLMS/Anubis#394]
Add endpoints for admin users (TAs/Profs) to be able to manually trigger the functions found here: 

https://github.com/AnubisLMS/Anubis/blob/main/api/anubis/lms/autograde.py

Then add a button to the frontend autograde pages for these endpoints to be hit.

**Comment by wabscale** [comment]
There is a bit more to this than just calling the function. The autograde functions generally take quite a while to run. We'll need the endpoints to enqueue the [bulk_autograde](https://github.com/AnubisLMS/Anubis/blob/main/api/anubis/lms/autograde.py#L151) function to run in an rpc queue.

**Comment by devbazregari** [comment]
sorry bro it's beyond my abilities. I haven't worked with RPC. but I know about celery and async if you can handle your work with these two techs I can be a little helpful

**Comment by wabscale** [comment]
We dont use celery, we use [python-rq](https://python-rq.org/). It is just like celery, but significantly simpler. I did half of the backend bit for this issue here if you want to see how it was put together. It was basically just making an enqueue function (something we do to keep our rpc calls organized) for the bulk_autograde.

https://github.com/AnubisLMS/Anubis/commit/ee32af19479b2290f435b01bc0888f2a7ed7b8ba

**Comment by dolf321** [comment]
Is somebody working on this(asking because its in progress on the board)

**Comment by wabscale** [comment]
No, no one is working on this right now. Go ahead if you would like to take it

**Comment by dolf321** [comment]
Yes, ill take it up thanks!

**Comment by dolf321** [comment]
Clarifying: 

You want an autograde button here:
![Image](https://user-images.githubusercontent.com/65448126/230644533-dc4174b9-185b-443a-b6c7-1d4cfcb0fa0a.png)

And you want me to call this "enqueue_bulk_autograde" function in the backend right, let me know if i'm missing anything here.

![Image](https://user-images.githubusercontent.com/65448126/230644740-ad94215d-3c90-4dec-b8b3-b82eadeafbe2.png)

**Comment by wabscale** [comment]
Oh the title of this is slightly misleading. We actually already have a button for regrading an entire assignment. We need a button for regrading all the submissions for a given student.

I guess it likely makes sense to put the actual button on the admin assignment page (the one with the assignment specific settings) that opens a [dialog](https://mui.com/material-ui/react-dialog/#customization) with an [autocomplete](https://mui.com/material-ui/react-autocomplete/) that has all the students. TA then selects the student from the dropdown, and clicks regrade. That hits the endpoint that will call the [enqueue_bulk_regrade_submissions](https://github.com/AnubisLMS/Anubis/blob/f680928dfabdeca17a1d6a2f8ad760fe1695289a/api/anubis/rpc/enqueue.py#L131) function with all the submissions from that student. Let me know if this makes sense.

You don't need to do the frontend if that is too much react/mui stuff. If you do half, I'll do the other half.

**Comment by dolf321** [comment]
Oh! Okay well I've definitely misconstrued that issue sorry(my poor button what a waste), yes i'll be on it. Hopefully will finish half of the frontend by today, would you like commit access to my fork or would you like me to pr unfinished frontend code?

**Comment by wabscale** [comment]
When you open the PR, I think you can check a box to allow maintainers (so me) to be able to push to the branch in your repo.

**Comment by dolf321** [comment]
@wabscale do you think it should go in the autograde results section instead?

**Comment by wabscale** [comment]
Well this action would completely invalidate everything on the student's results page. The new results will not be recalculated for at least a minute or two. So you would need to redirect after clicking somewhere else.

Feel free to add that too if you want, but the main way of doing it should be the way I described.
